### PR TITLE
download assets over https if no protocol is specified

### PR DESCRIPTION
**Pull request by mazlix**
#2926 #2927

### CODE REVIEW

**Comment by matthew-dean** [comment]
What happens if the https protocol is not valid?

**Comment by Synchro** [comment]
That's an implicit hazard of using anonymous/relative protocol URLs, which is one reason [they're not a good idea](http://stackoverflow.com/questions/4659345/is-there-any-downside-for-using-a-leading-double-slash-to-inherit-the-protocol-i/12899126#12899126). OTOH, if you're using them, you're expecting both HTTP and HTTPS to work for the same host anyway so it's unlikely to cause breakage. It used to be more of a problem than it is now - for example google used to serve from `http://www.google.com` and `https://secure.google.com`, so relative protocols would break, and though they don't do that any more, others might.

In short, use HTTPS by default, don't use anonymous protocol URLs.

**Comment by mazlix** [comment]
After reading a bit more on it, I do agree that it is probably a best practice to be explicit and use HTTPS rather than protocol-relative when switching from `@import` to `@import (inline)`, but I think this is a mistake others may make (since we just made it and got tons of mixed-content after deploy). 

I just think overall if we're going to assume either HTTP or HTTPS for downloading in a build-step we should opt for SSL since any absolute paths returned pointed at HTTP will throw mixed-content warnings, while if you're on HTTP and pointing to an HTTPS asset you'll have no such issue.

**Comment by matthew-dean** [comment]
> I just think overall if we're going to assume either HTTP or HTTPS for downloading in a build-step we should opt for SSL since any absolute paths returned pointed at HTTP will throw mixed-content warnings, while if you're on HTTP and pointing to an HTTPS asset you'll have no such issue.

You risk an error either way. In reality, protocol-relative URLs aren't really valid for compiling on the server, because they are relative to the protocol used in the HTML request, which doesn't exist at compile-time. Less is making a best-guess error-recovery decision, but, really, the URL should always be explicit, based on the target environment for your CSS file.

I suppose an error-recovery for a protocol-relative URL could try https first, and, on failing that, try HTTP. Buuuut.... if you really want protocol-relative behavior, and that's why you're not being explicit with your URL, then inlining content is probably not the answer. Nor is just switching the default protocol, as a URL is simply less likely to be valid as https.
